pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -0,0 +1,64 @@
+// Implements a check that provides detail on potentially faulty or customised
+// IngressController configurations on the default controller.
+//
+// Included checks are:
+//  - existence of custom ingress certificate
+//  - existence of default ingresscontroller
+
+package ingresscertificatechecker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ingressCertificateChecker interface {
+	Check(ctx context.Context) error
+}
+
+type checker struct {
+	operatorcli operatorclient.Interface
+	configcli   configclient.Interface
+}
+
+func newIngressCertificateChecker(operatorcli operatorclient.Interface, configcli configclient.Interface) *checker {
+	return &checker{
+		operatorcli: operatorcli,
+		configcli:   configcli,
+	}
+}
+
+func (r *checker) Check(ctx context.Context) error {
+	cv, err := r.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	ingress, err := r.operatorcli.OperatorV1().IngressControllers("openshift-ingress-operator").Get(ctx, "default", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if ingress.Spec.DefaultCertificate == nil {
+		// TODO: fix constant reconciliation.
+		// This causes constant reconciliation in dev where we do not have default cert set.
+		// In the current implementation we seem to always set condition to False in dev,
```

Comment:
I think we also set it to false when there is a custom domain in use because of this bit:

https://github.com/Azure/ARO-RP/blob/340ebb161f590eaf4ee069d519e6cfdd9d5c5f01/pkg/cluster/tls.go#L29-L36